xwiki-rendering-syntaxes/xwiki-rendering-syntax-annotatedxhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/AnnotatedXHTMLChainingRenderer.java

```diff
@@ -83,4 +86,37 @@ public void endMacroMarker(String name, Map<String, String> parameters, String c
             this.macroRenderer.endRender(getXHTMLWikiPrinter());
         }
     }
+
+    /**
+     * @return a span element if we are inside an inline macro. Else a div.
+     */
+    private String getMetadataContainerElement()
+    {
+        if (getBlockState().isInLine()) {
+            return "span";
+        } else {
+            return "div";
+        }
+    }
+
+    @Override
+    public void beginMetaData(MetaData metadata)
+    {
+        Map<String, String> attributes = new LinkedHashMap<>();
+
+        for (Map.Entry<String, Object> metadaPair : metadata.getMetaData().entrySet()) {
```

Comment:
``metada``?

Reply:
:D I took 5 minutes to understand your question, apparently my brain reads directly `metadata` from `metada` now ^^